-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39532/32317

ERROR: Build errors found during clang-tidy run.
```
                                             typename 
--
DataFormats/SoATemplate/interface/SoACommon.h:506:46: error: missing 'typename' prior to dependent type name 'Matrix<double, 3, 6, 0>::Scalar' [clang-diagnostic-error]
    static constexpr auto valueSize = sizeof(C::Scalar);
                                             ^
                                             typename 
--
DataFormats/SoATemplate/interface/SoACommon.h:506:46: error: missing 'typename' prior to dependent type name 'const Matrix<double, 3, 6, 0>::Scalar' [clang-diagnostic-error]
    static constexpr auto valueSize = sizeof(C::Scalar);
                                             ^
                                             typename 
--
DataFormats/SoATemplate/interface/SoACommon.h:506:46: error: missing 'typename' prior to dependent type name 'Matrix<double, 3, 6, 0>::Scalar' [clang-diagnostic-error]
    static constexpr auto valueSize = sizeof(C::Scalar);
                                             ^
                                             typename 
--
DataFormats/SoATemplate/interface/SoACommon.h:506:46: error: missing 'typename' prior to dependent type name 'const Matrix<double, 3, 6, 0>::Scalar' [clang-diagnostic-error]
    static constexpr auto valueSize = sizeof(C::Scalar);
                                             ^
                                             typename 
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/DataFormats/PortableTestObjects/BuildFile.xml
+++ b/DataFormats/PortableTestObjects/BuildFile.xml
@@ -2,6 +2,7 @@
 <use name="eigen"/>
 <use name="DataFormats/Common"/>
 <use name="DataFormats/Portable"/>
+<use name="DataFormats/SoATemplate"/>
 <use name="HeterogeneousCore/AlpakaInterface"/>
 <flags ALPAKA_BACKENDS="1"/>
 <export>

--- a/DataFormats/SoATemplate/interface/SoACommon.h
+++ b/DataFormats/SoATemplate/interface/SoACommon.h
@@ -390,7 +390,7 @@ namespace cms::soa {
     }
 
     using ValueType = typename C::Scalar;
-    static constexpr auto valueSize = sizeof(C::Scalar);
+    static constexpr auto valueSize = sizeof(typename C::Scalar);
     SOA_HOST_DEVICE SOA_INLINE byte_size_type stride() const { return stride_; }
 
   private:
@@ -503,7 +503,7 @@ namespace cms::soa {
     SOA_HOST_DEVICE SOA_INLINE const C* operator&() const { return &cVal_; }
 
     using ValueType = typename C::Scalar;
-    static constexpr auto valueSize = sizeof(C::Scalar);
+    static constexpr auto valueSize = sizeof(typename C::Scalar);
 
     SOA_HOST_DEVICE SOA_INLINE byte_size_type stride() const { return stride_; }
 

